src/core/lib/promise/latch.h

```diff
@@ -77,6 +79,25 @@ class Latch {
     };
   }
 
+  // Produce a promise to wait for a value from this latch.
+  // Copies the result out of the latch.
+  auto WaitAndCopy() {
```

Comment:
Please make sure there's a test for this.

Reply:
done